### ISSUE TITLE
Monitored metric (val loss) not found when resuming from checkpoint

### ISSUE THREAD

**Issue by samgelman** [PyTorchLightning/pytorch-lightning#12923]
## 🐛 Bug

I get `UserWarning: ModelCheckpoint(monitor='val_loss') could not find the monitored key in the returned metrics: ['epoch', 'step']` when resuming training from checkpoint. However, I am logging val_loss, so I would expect it to be there. I think the problem is related to specifying a custom dirpath and log directory, but I'm not sure.

### To Reproduce

https://colab.research.google.com/drive/1yZXNpfBJ9e8TPgAroMwdi-m99g1JKLpe?usp=sharing

### Expected behavior

Lightning should be able to find val_loss

### Environment

Lightning 1.6.2 (see link above)

### Additional context

cc @carmocca @awaelchli @ninginthecloud @jjenniferdai @rohitgr7
